spec/controllers/councillor_contributions_controller_spec.rb

```diff
@@ -0,0 +1,23 @@
+require 'spec_helper'
+
+describe CouncillorContributionsController do
+    let(:authority){create(:authority)}
+    let(:councillor_contribution){create(:councillor_contribution, authority: authority)}
+    let(:suggested_councillors){create(:suggested_councillor, councillor_contribution: councillor_contribution)}
+
+  context "when the feature flag is on" do
+    around do |test|
+      with_modified_env CONTRIBUTE_COUNCILLORS_ENABLED: "true" do
+        test.run
+      end
+    end
+
+    describe "#show" do
+      it "download csv file" do
+        get :show, authority_id: authority.id, id: councillor_contribution.id, format: "csv"
+        expect(response.header['Content-Type']).to include'text/csv'
+        expect(response.body).to include "name,start_date,end_date,exective,council,council_website,id,email,image,party,source,ward,phone_mobile"
```

Comment:
This doesn't currently check that the SuggestedCouncillor we've set up is available.

What do you think of actually running CSV.parse http://ruby-doc.org/stdlib-2.4.1/libdoc/csv/rdoc/CSV.html#method-c-parse and then `headers` and something like `to_hash` on the first row to check that the councillor data is there. Something like:

```
response_csv = CSV.parse(response.body)

expect((response_csv).headers).to eql 'that array we want'
expect((response_csv.read.last.to_s).to eql 'the array of values we expect'
```

Here's the docs for the CSV::Row class https://ruby-doc.org/stdlib-2.4.1/libdoc/csv/rdoc/CSV/Row.html#method-i-to_s

And CSV https://ruby-doc.org/stdlib-2.4.1/libdoc/csv/rdoc/CSV.html

The advantage of this is that we're also testing that we're creating a readable CSV, which is pretty important :)